kie-maven-plugin/src/main/java/org/kie/maven/plugin/GenerateDMNModelMojo.java

```diff
@@ -94,22 +99,14 @@ private void generateDMNModel() throws MojoExecutionException {
 
             DMNCompilerConfigurationImpl dmnCompilerConfiguration = (DMNCompilerConfigurationImpl) DMNFactory.newCompilerConfiguration();
 
-            final List<String> compiledClassNames = new ArrayList<>();
+            Map<String, String> classNameSourceMap = new HashMap<>();
+
             dmnCompilerConfiguration.setDeferredCompilation(true);
             dmnCompilerConfiguration.addListener(generatedSource -> {
-                final String droolsModelCompilerPath = "/generated-sources/dmn/main/java";
-                addNewCompileRoot(droolsModelCompilerPath);
-
                 for (GeneratedSource generatedFile : generatedSource) {
                     final Path fileNameRelative = transformPathToMavenPath(generatedFile);
-
-                    compiledClassNames.add(getCompiledClassName(fileNameRelative));
-
-                    final Path newFile = Paths.get(targetDirectory.getPath(),
-                                                   droolsModelCompilerPath,
-                                                   fileNameRelative.toString());
-
-                    createInvokerSourceFile(newFile, generatedFile.getSourceContent());
+                    getLog().info("Generating new DMN file: " + generatedFile);
+                    classNameSourceMap.put(getCompiledClassName(fileNameRelative), generatedFile.getSourceContent());
```

Comment:
Are you sure this code is equivalent to the old one?
I see for example the code of method `createDMNFile` and I'm not sure the new code is equivalent

Reply:
Sorry, I'm not sure what's the problem here. I'm actually replacing that `createInvokerSourceFile` method that dumped the source file with mine that put it in the map. Are you confusing the methods? Am I missing something?